core/repository/http/src/main/java/org/eclipse/rdf4j/repository/http/HTTPRepositoryConnection.java

```diff
@@ -658,6 +660,10 @@ protected void removeWithoutCommit(Resource subject, IRI predicate, Value object
 		if (object == null) {
 			object = SESAME.WILDCARD;
 		}
+		for (int i = 0; i < contexts.length; i++) {
+			if (contexts[i] == null)
+				contexts[i] = RDF4J.NIL;
+		}
```

Comment:
This looks like the right fix to me (I could have sworn we have a util function to do exactly this swap, but can't find it immediately).